core/controllers/editor.py

```diff
@@ -481,6 +482,68 @@ def put(self, exploration_id):
         })
 
 
+class ExplorationModeratorRightsHandler(EditorHandler):
+    """Handles management of exploration rights by moderators."""
+
+    PAGE_NAME_FOR_CSRF = 'editor'
+
+    @base.require_moderator
+    def put(self, exploration_id):
+        """Updates the editing rights for the given exploration, and sends an
+        email to all its owners.
+        """
+        exploration = exp_services.get_exploration_by_id(exploration_id)
+        action = self.payload.get('action')
+        email_body = self.payload.get('email_body')
+        version = self.payload.get('version')
+        _require_valid_version(version, exploration.version)
+
+        if action not in feconf.VALID_MODERATOR_ACTIONS:
+            raise self.InvalidInputException('Invalid moderator action.')
+
+        # If moderator emails can be sent, check that all the prerequisites are
+        # satisfied, otherwise do nothing.
+        if feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION:
+            if not email_body:
```

Comment:
This was an error, thanks for catching it. I used a HTML field when it should have been a unicode field (sometimes the former inserts stuff like '<p></p>').